cupy/fft/fft.py

```diff
@@ -175,6 +185,47 @@ def _fft(a, s, axes, norm, direction, value_type='C2C', overwrite_x=False,
     return a
 
 
+def _prep_fftn_axes(ndim, s=None, axes=None):
+    """Configure axes argument for an n-dimensional FFT.
+
+    The axes to be transformed are returned in ascending order.
+    """
+
+    # compatibility checks for cupy.cuda.cufft.PlanNd
+    if (s is not None) and (axes is not None) and len(s) != len(axes):
+        raise ValueError("Shape and axes have different lengths.")
+
+    if axes is None:
+        if s is None:
+            dim = ndim
+        else:
+            dim = len(s)
+        axes = tuple([i + ndim for i in six.moves.range(-dim, 0)])
+        axes_sorted = axes
+    else:
+        axes = tuple(axes)
+        if axes is ():
+            return (), ()
+        if _reduce(min, axes) < -ndim or _reduce(max, axes) > ndim - 1:
+            raise ValueError("The specified axes exceed the array dimensions.")
+        axes_sorted = tuple(sorted([ax % ndim for ax in axes]))
+
+    # unsorted axes for _cook_shape, sorted ones are otherwise used
+    return axes, axes_sorted
+
+
+def _nd_plan_is_possible(axes_sorted, ndim):
+    # PlanNd supports 1D, 2D and 3D batch transforms over contiguous axes
+    if (len(axes_sorted) > 3 or len(axes_sorted) == 0 or
+            # PlanNd only possible if the first or last axis is in axes.
+            ((0 not in axes_sorted) and ((ndim - 1) not in axes_sorted)) or
+            # axes are not contiguous
+            not all([(axes_sorted[n + 1] - axes_sorted[n]) == 1
+                     for n in range(len(axes_sorted) - 1)])):
+        return False
+    return True
```

Comment:
- Do you want to keep the error message for `if len(fft_axes) < 1 or len(fft_axes) > 3`?
- How about the following?
```python
    return (
        0 < len(axes_sorted) <= 3
        and (0 in axes_sorted or (ndim - 1) in axes_sorted)
        and all([
            (axes_sorted[n + 1] - axes_sorted[n]) == 1
            for n in range(len(axes_sorted) - 1)]))
```

Reply:
This form looks better, thanks. As far as the other error message you mention, I think we should remove it and just mention 1-3 axes in the case that `_nd_plan_is_possible` returns False.